airbyte-webapp/src/components/connectorBuilder/Builder/BuilderCard.tsx

```diff
@@ -33,7 +52,139 @@ export const BuilderCard: React.FC<React.PropsWithChildren<BuilderCardProps>> =
+            <CopyToModal
+              onCancel={() => {
+                setCopyToOpen(false);
+              }}
+              onApply={(selectedStreamIndices) => {
+                const sectionToCopy = getFieldMeta(
+                  `streams[${copyConfig.currentStreamIndex}].${copyConfig.path}`
+                ).value;
+                selectedStreamIndices.forEach((index) => {
+                  setFieldValue(`streams[${index}].${copyConfig.path}`, sectionToCopy, true);
+                });
+                setCopyToOpen(false);
+              }}
+              currentStreamIndex={copyConfig.currentStreamIndex}
+              title={copyConfig.copyToLabel}
+            />
+          )}
+          <Button
+            variant="secondary"
+            type="button"
+            onClick={() => {
+              setCopyFromOpen(true);
+            }}
+            icon={<FontAwesomeIcon icon={faArrowUpRightFromSquare} rotation={180} />}
+          />
+          {isCopyFromOpen && (
+            <CopyFromModal
+              onCancel={() => {
+                setCopyFromOpen(false);
+              }}
+              onSelect={(selectedStreamIndex) => {
+                setFieldValue(
+                  `streams[${copyConfig.currentStreamIndex}].${copyConfig.path}`,
+                  getFieldMeta(`streams[${selectedStreamIndex}].${copyConfig.path}`).value,
+                  true
+                );
+                setCopyFromOpen(false);
+              }}
+              currentStreamIndex={copyConfig.currentStreamIndex}
+              title={copyConfig.copyFromLabel}
+            />
+          )}
+        </div>
+      )}
       {(!toggleConfig || toggleConfig.toggledOn) && children}
     </Card>
   );
 };
+
+const CopyToModal: React.FC<{
+  onCancel: () => void;
+  onApply: (selectedStreamIndices: number[]) => void;
+  title: string;
+  currentStreamIndex: number;
+}> = ({ onCancel, onApply, title, currentStreamIndex }) => {
+  const [streams] = useField<BuilderStream[]>("streams");
+  const [selectMap, setSelectMap] = useState<Record<string, boolean>>({});
+  return (
+    <Modal size="sm" title={title} onClose={onCancel}>
+      <form
+        onSubmit={() => {
+          onApply(
+            Object.entries(selectMap)
+              .filter(([, selected]) => selected)
+              .map(([index]) => Number(index))
+          );
+        }}
+      >
+        <ModalBody className={styles.modalStreamListContainer}>
+          {streams.value.map((stream, index) => (
+            <label htmlFor={`copy-to-stream-${index}`} key={index} className={styles.toggleContainer}>
+              <CheckBox
+                id={`copy-to-stream-${index}`}
+                checked={selectMap[index] || false}
+                disabled={index === currentStreamIndex}
+                onChange={() => {
+                  setSelectMap({ ...selectMap, [index]: !selectMap[index] });
+                }}
+              />
+              {stream.name}
```

Comment:
small nit: if a stream does not have a name, then the text is just blank in both of these modals. Maybe we should show something like `(empty)` in this case, similar to how we do in the stream sidebar?